packages/remix-react/browser.tsx

```diff
@@ -213,48 +218,56 @@ export function RemixBrowser(_props: RemixBrowserProps): ReactElement {
       window.__remixManifest.routes,
       window.__remixRouteModules,
       window.__remixContext.state,
-      window.__remixContext.future
+      window.__remixContext.future,
+      window.__remixContext.isSpaMode
     );
 
-    // Create a shallow clone of `loaderData` we can mutate for partial hydration.
-    // When a route exports a `clientLoader` and a `HydrateFallback`, the SSR will
-    // render the fallback so we need the client to do the same for hydration.
-    // The server loader data has already been exposed to these route `clientLoader`'s
-    // in `createClientRoutes` above, so we need to clear out the version we pass to
-    // `createBrowserRouter` so it initializes and runs the client loaders.
-    let hydrationData = {
-      ...window.__remixContext.state,
-      loaderData: { ...window.__remixContext.state.loaderData },
-    };
-    let initialMatches = matchRoutes(routes, window.location);
-    if (initialMatches) {
-      for (let match of initialMatches) {
-        let routeId = match.route.id;
-        let route = window.__remixRouteModules[routeId];
-        let manifestRoute = window.__remixManifest.routes[routeId];
-        // Clear out the loaderData to avoid rendering the route component when the
-        // route opted into clientLoader hydration and either:
-        // * gave us a HydrateFallback
-        // * or doesn't have a server loader and we have no data to render
-        if (
-          route &&
-          shouldHydrateRouteLoader(manifestRoute, route) &&
-          (route.HydrateFallback || !manifestRoute.hasLoader)
-        ) {
-          hydrationData.loaderData[routeId] = undefined;
-        } else if (manifestRoute && !manifestRoute.hasLoader) {
-          // Since every Remix route gets a `loader` on the client side to load
-          // the route JS module, we need to add a `null` value to `loaderData`
-          // for any routes that don't have server loaders so our partial
-          // hydration logic doesn't kick off the route module loaders during
-          // hydration
-          hydrationData.loaderData[routeId] = null;
+    let hydrationData = undefined;
+    if (!window.__remixContext.isSpaMode) {
```

Comment:
This is just a whitespace change - we don't have to prep any `hydrationData` in SPA mode